src/main/kotlin/org/janelia/saalfeldlab/labels/blocks/n5/LabelBlockLookupFromN5.kt

```diff
@@ -75,83 +81,95 @@ class LabelBlockLookupFromN5(
 		}
 	}
 
-	private val attributes = mutableMapOf<Int, DatasetAttributes>()
-
-
-	@Throws(IOException::class)
-	fun set(level: Int, ids: Map<Long, Array<Interval>>) {
-
-
-		val dataset = String.format(scaleDatasetPattern, level)
-		val attributes = this.attributes.getOrPut(level, { n5().getDatasetAttributes(dataset) })
-		val stepSize = attributes.blockSize[0]
-
-		val mapByBlock = mutableMapOf<Long, MutableMap<Long, Array<Interval>>>()
-
-		for (entry in ids)
-			mapByBlock.computeIfAbsent((entry.key / stepSize) * stepSize, { mutableMapOf() })[entry.key] = entry.value
-
-		for (m in mapByBlock)
-			writeMap(level, m.key, m.value)
-
+	override fun read(key: LabelBlockLookupKey): Array<Interval> {
+		val map = cache.get(getBlockKey(key), this::readBlock)
+		return map[key.id] ?: arrayOf()
 	}
 
+	override fun write(key: LabelBlockLookupKey, vararg intervals: Interval) {
+		val blockKey = getBlockKey(key)
+		cache.invalidate(blockKey)
+		val map = readBlock(blockKey).toMutableMap()
+		map[key.id] = arrayOf(*intervals)
+		writeBlock(getBlockKey(key), map)
```

Comment:
Good point about concurrent access. This one is a bit tricky, so for now I just made all public methods synchronized. The access is not too frequent so it shouldn't hurt, and I tested it with a selection of ~8500k objects and didn't notice any performance implications (the difference is just a few ms for the entire set according to my benchmark).

Otherwise I find it hard to reason about how it would work without synchronization, since label ids somewhat depend on each other if they are in the same block (some individual writes may get lost, for example).